src/EPub.ts

```diff
@@ -0,0 +1,244 @@
+        .then((str) => new window.DOMParser().parseFromString(str, "text/xml"))
+        .then((data) => JSON.stringify(xmlToJson(data)));
+
+      if (store) {
+        await store.set("manifest", JSON.stringify(manifestJSON));
+      }
+
+      return new Manifest(JSON.parse(manifestJSON), manifestUrl);
+    };
+
+    const tryToUpdateManifestButIgnoreResult = async (): Promise<void> => {
+      try {
+        await fetchManifest();
+      } catch (err) {
+        console.log("Something went wrong", err);
+      }
+      return new Promise<void>((resolve) => resolve());
+    };
+
+    // Respond immediately with the manifest from the store, if possible.
+    if (store) {
+      const manifestString = await store.get("manifest");
+      if (manifestString) {
+        // Kick off a fetch to update the store for next time,
+        // but don't await it.
+        tryToUpdateManifestButIgnoreResult();
+        const manifestJSON = JSON.parse(manifestString);
+        return new Manifest(manifestJSON, manifestUrl);
+      }
+    }
+
+    return fetchManifest();
+  }
+
+  public parseMetaData(manifestJSON: any): any {
+    return manifestJSON?.package?.metadata
+      ? {
+          title: manifestJSON.package.metadata["dc:title"]["#text"],
+        }
+      : {};
+  }
+
+  public parseTOC(manifestJSON: any): any {
+    const emptySpine: string[] = [];
+
+    return (manifestJSON?.package?.manifest?.item || emptySpine).reduce(
+      (acc: any, chapter: { href: string; id: string }) => {
+        acc.push({
+          //@ts-ignore
+          href: chapter["@attributes"]["href"],
+          //@ts-ignore
+          title: chapter["@attributes"]["id"],
+        });
+        return acc;
+      },
+      []
+    );
+  }
+  public parseSpine(manifestJSON: any): any {
+    const emptySpine: string[] = [];
+    return (manifestJSON?.package?.spine?.itemref || emptySpine).reduce(
+      (acc: any, chapter: { idref: string; linear: string }) => {
+        acc.push({
+          type: "application/xhtml+xml",
+          href: manifestJSON?.package?.manifest?.item.filter(
+            //@ts-ignore
+            (item: any) =>
+              //@ts-ignore
+              item["@attributes"]["id"] === chapter["@attributes"]["idref"] &&
+              item["@attributes"]["href"]
+          )[0]["@attributes"]["href"],
+        });
+        return acc;
+      },
+      []
+    );
+  }
+
+  public parseResources(manifestJSON: any): any {
+    return (manifestJSON?.package?.manifest?.item || []).reduce(
+      (acc: any, current: any) => {
+        acc.push({
+          href: current["@attributes"]["href"],
+          id: current["@attributes"]["id"],
+        });
+        return acc;
+      },
+      []
+    );
+  }
+  public constructor(manifestJSON: any, manifestUrl: URL) {
+    this.metadata = this.parseMetaData(manifestJSON);
+    //links format should be updated to point to manifest.json
+    this.links = this.parseTOC(manifestJSON);
```

Comment:
```{
        href: manifestJSON.href,
        type: "application/webpub+json",
        templated: false,
        rel: "self",
      },``` This should look something like .. but it's supposed to point to a _manifest.json_ file. I am not sure if links are optional for the e-reader?

Reply:
Aren't the links the chapter links for the TOC?  That seems like an important thing ... but I might be mistaken

Reply:
@hokei I initially thought links was another method for parsing TOC but the standard is for link to contain links to the manifest.json. That's how the other EPub in the static folder is formatting it's link (linking to manifest.json)

The spec is viewable here: https://github.com/readium/webpub-manifest#23-links 
example: 
```
"links": [
    {"rel": "self", "href": "https://example.com/manifest.json", "type": "application/webpub+json"},
    {"rel": "alternate", "href": "https://example.com/publication.epub", "type": "application/epub+zip"},
    {"rel": "search", "href": "https://example.com/search{?query}", "type": "text/html", "templated": true}
  ]
```